File: keyly-extension/src/models/Compartit.ts
import type { Item } from "./Item"

export type Compartit = {
  uuid: string
  usuariCreador: {
    uuid: string
    nom: string
    correu: string
  },
    usuariReceptor: {
        uuid: string
        nom: string
        correu: string
  }
  tipusEntitat: TipusEntitat
  permisos: Permisos
  carpeta?: CompartitCarpeta
  item?: Item
  dataCreacio: string
}

type CompartitCarpeta = {
  uuid: string
  nom: string
  favorit: boolean
  items: Item[]
}
export type CompartitRequest = {
  entitatUuid: string
  tipusEntitat: TipusEntitat
  usuaris: {
    usuariUuid: string
    permis: Permisos
    encryptedDataKey?: string
  }[]
}

export type CompartitItemRequest = {
  itemRequest: Item
  compartitRequest: CompartitRequest
}

type CompartitItem = {
  uuid: string
  titol: string
  nomUsuari: string
  contrasenya: string
  iv: string
  encryptedDataKey: {
    uuid: string
    encryptedDatakey: string
  }
  url: string
  notes: string
  favorit: boolean
  dinsDeCarpeta: boolean
}

export enum TipusEntitat {
  CARPETA = "CARPETA",
  ITEM = "ITEM"
}

export enum Permisos {
  LECTURA = "LECTURA",
  ESCRIPTURA = "ESCRIPTURA",
  ADMINISTRADOR = "ADMINISTRADOR"
}

File: Android/app/src/main/java/com/example/keyly_projecte_intermodular/dao/Compartit.java
package com.example.keyly_projecte_intermodular.dao;

import com.example.keyly_projecte_intermodular.utils.TipusEntitat;
import com.example.keyly_projecte_intermodular.utils.Permisos;

import java.io.Serializable;
import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Compartit implements Serializable {
    private UUID uuid;
    private Usuari usuariCreador;
    private Usuari usuariReceptor;
    private TipusEntitat tipusEntitat;
    private Permisos permisos;
    private Carpeta carpeta;
    private Item item;
    private String dataCreacio;
    private int comptadorAccess;
    private String ultimAccess;
}
